### PR TITLE
Intuitively implement properties of ChorusEffect

### LINKED ISSUE #89: Type and values for some properties in ReverbEffect and ChorusEffect are not intuitive
According to [OpenAL doc](https://kcat.strangesoft.net/misc-downloads/Effects%20Extension%20Guide.pdf), `AL_EAXREVERB_DECAYHF_LIMIT` has only two values, `AL_TRUE` and `AL_FALSE`. However, the current type for it is `int`, probably because in C it makes more sense to have int. However, here in Python, I believe it would makes more sense to have it as a boolean property.

### PR DESCRIPTION
**Pull request by 9a24f0**
Opened PR for better tracking.

### CODE REVIEW

File: src/palace.pyx [hunk 1]
```diff
@@ -2338,9 +2338,10 @@ cdef class ChorusEffect(BaseEffect):
     """
     cdef alure.EFXCHORUSPROPERTIES properties
 
-    def __init__(self, waveform: int, phase: int,
-                 depth: float, feedback: float, delay: float,
-                 context: Optional[Context] = None) -> None:
+    def __init__(self, waveform: str = 'triangle',
+                 phase: int = 90, depth: float = 0.1,
+                 feedback: float = 0.25, delay: float,
+                 context: Optional[Context] = None) -> None:
```
Comment (Huy-Ngo) [inline src/palace.pyx]: Flake8 complains about trailing whitespace on 2344

File: src/palace.pyx [hunk 2]
```diff
@@ -2351,12 +2352,18 @@ cdef class ChorusEffect(BaseEffect):
...
+            return 'sin'
 
     @waveform.setter
-    def waveform(self, value: int) -> None:
-        self.properties.waveform = value
+    def waveform(self, value: str) -> None:
+        if value == 'triangle':
+            self.properties.waveform = 1
+        else:
+            self.properties.waveform = 0;
```
Comment (Huy-Ngo) [inline src/palace.pyx]: There should be no semicolon here.

File: src/palace.pyx [hunk 3]
```diff
@@ -2338,9 +2338,10 @@ cdef class ChorusEffect(BaseEffect):
     """
     cdef alure.EFXCHORUSPROPERTIES properties
 
-    def __init__(self, waveform: int, phase: int,
-                 depth: float, feedback: float, delay: float,
-                 context: Optional[Context] = None) -> None:
+    def __init__(self, waveform: str = 'triangle',
+                 phase: int = 90, depth: float = 0.1,
+                 feedback: float = 0.25, delay: float,
```
Comment (Huy-Ngo) [inline src/palace.pyx]: Argument `delay` should have a default value (0.16) too.

File: src/palace.pyx [hunk 4]
```diff
@@ -2338,9 +2338,10 @@ cdef class ChorusEffect(BaseEffect):
...
+                 context: Optional[Context] = None) -> None:    
         super().__init__(context)
         self.properties.waveform = waveform
```
Comment (Huy-Ngo) [inline src/palace.pyx]: 'Here the input should be int, I think.

**Review by Huy-Ngo — CHANGES_REQUESTED** [review]
Here are some few mistakes that blocks tests from passing.

**Comment by McSinyx** [comment]
I'd prefer the PR to be small to be easier to review, so keep the scope within ChorusEffect.  If it goes fast, we can do ReverbEffect before the weekly release, otherwise there's no need to rush.

File: src/palace.pyx [hunk 5]
```diff
@@ -2307,11 +2307,11 @@ cdef class ReverbEffect(BaseEffect):
         self.slot.apply_effect(self.impl)
 
     @property
-    def decay_hf_limit(self) -> int:
+    def decay_hf_limit(self) -> bool:
         return self.properties.decay_hf_limit
 
     @decay_hf_limit.setter
-    def decay_hf_limit(self, value: int) -> None:
+    def decay_hf_limit(self, value: bool) -> None:
```
Comment (McSinyx) [inline src/palace.pyx:2314]: Genuinely asking, how is typing enforced by these hints, i.e. what's `type(ReverbEffect().decay_hf_limit)` initially and after, say, setting `.decay_hf_limit = []`?  IMHO it's OK to have boolean variable to be set to anything, long as the functionality works.

Though this one belong to another PR.

File: src/palace.pyx [hunk 6]
```diff
@@ -2322,7 +2322,7 @@ cdef class ChorusEffect(BaseEffect):
 
     Parameters
     ----------
-    waveform : int
+    waveform : str
     phase : int
     depth : float
     feedback : float
```
Comment (McSinyx) [inline src/palace.pyx]: Please add description for these values based on the docs, since we're polishing things here.

File: src/palace.pyx (hunk 3 above)
Comment (McSinyx) [inline src/palace.pyx]: What's the default value for depth?

File: src/palace.pyx [hunk 7]
```diff
@@ -2351,12 +2352,18 @@ cdef class ChorusEffect(BaseEffect):
         self.slot.apply_effect(self.impl)
 
     @property
-    def waveform(self) -> int:
-        return self.properties.waveform
+    def waveform(self) -> str:
+        if self.properties.waveform == 1:
+            return 'triangle'
+        else:
+            return 'sin'
```
Comment (McSinyx) [inline src/palace.pyx]:
```suggestion
        return 'triangle' if self.properties.waveform else 'sine'
```

We need docstring here too.

File: src/palace.pyx (hunk 2 above)
Comment (McSinyx) [inline src/palace.pyx]:
```suggestion
        elif value == 'sine':
            self.properties.waveform = 0
        else:
            raise ValueError(message here)
```

File: tests/unit/test_effect.py [hunk 8]
```diff
@@ -403,17 +402,14 @@ def test_reverb_decay_hf_limit(context):
...
-        assert isclose(fx.waveform, 1)
-        fx.rate = 0
-        assert fx.rate == 0
-        fx.rate = 1
-        assert fx.rate == 1
-        with raises(ValueError): fx.rate = 2
-        with raises(ValueError): fx.rate = -1
+        assert fx.waveform == 'triangle'
+        fx.waveform = 'sin'
+        assert fx.waveform == 'sin'
```
Comment (McSinyx) [inline tests/unit/test_effect.py]: Use `sine` please, also please check for invalid values.

**Review by McSinyx — CHANGES_REQUESTED** [review]

File: src/palace.pyx (hunk 3 above)
Comment (Huy-Ngo) [reply src/palace.pyx]: It's 0.1

Comment (McSinyx) [reply src/palace.pyx]: delay damn my dyslexium, also late to the party

File: src/palace.pyx (hunk 4 above)
Comment (McSinyx) [reply src/palace.pyx]: I think the better way to do it is to set `self.waveform`.

File: src/palace.pyx (hunk 2 above)
Comment (9a24f0) [reply src/palace.pyx]: LOL I was staring and having no idea where the semi colon at. Here is how it look on my editor
Screenshot from 2020-04-28 19-46-54

Comment (Huy-Ngo) [reply src/palace.pyx]: Lol that's because the color highlighting for semicolon is too so similar to the background.

**Comment by Huy-Ngo** [comment]
Btw, *intuitively, you made a typo on the PR name.

File: src/palace.pyx [hunk 9]
```diff
@@ -2338,10 +2338,10 @@ cdef class ChorusEffect(BaseEffect):
     """
     cdef alure.EFXCHORUSPROPERTIES properties
 
-    def __init__(self, waveform: str = 'triangle',
+    def __init__(self, waveform: int = 1,
```
Comment (McSinyx) [inline src/palace.pyx]: It was fine before lol

Comment (9a24f0) [reply src/palace.pyx]: Wut, it said that it needs an int before o.O

File: src/palace.pyx [hunk 10]
```diff
@@ -2351,12 +2352,17 @@ cdef class ChorusEffect(BaseEffect):
...
     @waveform.setter
-    def waveform(self, value: int) -> None:
-        self.properties.waveform = value
+    def waveform(self, value: str) -> None:
+        if value == 'triangle':
+            self.properties.waveform = 1
+        elif value == 'sine':
+            self.properties.waveform = 0
+        else:
+            raise ValueError(f'invalid waveform value')
```
Comment (McSinyx) [inline src/palace.pyx]:
```suggestion
            raise ValueError(f'invalid waveform: {value}')
```

File: src/palace.pyx [hunk 11]
```diff
@@ -2338,11 +2339,12 @@ cdef class ChorusEffect(BaseEffect):
...
+    def __init__(self, waveform: str = 'triangle',
+                 phase: int = 90, depth: float = 0.1,
+                 feedback: float = 0.25, delay: float = 0.016,
                  context: Optional[Context] = None) -> None:
         super().__init__(context)
-        self.properties.waveform = waveform
+        self.waveform = waveform
         self.properties.phase = phase
         self.properties.depth = depth
         self.properties.feedback = feedback
```
Comment (McSinyx) [inline src/palace.pyx]: Make it for others too.

File: src/palace.pyx [hunk 12]
```diff
@@ -2334,6 +2335,8 @@ cdef class ChorusEffect(BaseEffect):
 
     Raises
     ------
+    ValueError
+        If the waveform is neither set to 'sine' nor 'triangle'.
```
Comment (McSinyx) [inline src/palace.pyx]: Not sure if we want this, feeling it's a bit redundant.  @Huy-Ngo, what do you think?

File: src/palace.pyx [hunk 13]
```diff
@@ -2323,7 +2323,8 @@ cdef class ChorusEffect(BaseEffect):
     Parameters
     ----------
     waveform : str
-        
+        Either 'sine' or 'triangle'.
+        By default 'triangle' is used.
```
Comment (McSinyx) [inline src/palace.pyx]: Sorry was forgetting that the default value is in the signature of `__init__`, i.e. we don't need the second sentence.

**Review by McSinyx — CHANGES_REQUESTED** [review]

File: src/palace.pyx (hunk 12 above)
Comment (Huy-Ngo) [reply src/palace.pyx]: Yea I think it's redundant.

File: src/palace.pyx [hunk 14]
```diff
@@ -2324,7 +2324,6 @@ cdef class ChorusEffect(BaseEffect):
     ----------
     waveform : str
         Either 'sine' or 'triangle'.
-        By default 'triangle' is used.
     phase : int
     depth : float
     feedback : float
```
Comment (McSinyx) [inline src/palace.pyx]: Add the `From -180 to 180.` things here too.

**Review by McSinyx — APPROVED** [review]
LGTM, if Travis agrees.  You may want to lift `xfail` for tests for `ChorusEffect`.

**Comment by 9a24f0** [comment]
> You may want to lift `xfail` for tests for `ChorusEffect`.

Other failed because the it didn't raise :/
